sqrl-planner/sqrl-common/src/main/java/com/datasqrl/calcite/SqrlToSql.java

```diff
@@ -436,20 +463,23 @@ private Optional<String> getIdentifier(SqlNode item) {
     return Optional.empty();
   }
 
+  Optional<SqlNode> condition = Optional.empty();
```

Comment:
It seems out of place to me to have this condition has a variable in a nested visitor pattern. Should this be a stack? What happens in deeply nested queries?

Reply:
This is for one specific case where we eliminate a table for join declarations. e.g.`X := JOIN ON condition`, the condition gets pulled into the where clause to make `FROM X WHERE condition`. There's a comment in the code about this. I'll put it in the stack but it becomes a bit more complex.